cpp/cmake/bob.cmake

```diff
@@ -189,76 +263,38 @@ function(bob_begin_cxx_flags)
       set(FLAGS "${FLAGS} -g")
     endif()
     # -----------
-    if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-      if(${PROJECT_NAME}_CXX_WARNINGS)
-        set(FLAGS "${FLAGS} -Werror -Weverything")
-        set(FLAGS "${FLAGS} -Wno-documentation-unknown-command")
-        set(FLAGS "${FLAGS} -Wno-padded")
-        set(FLAGS "${FLAGS} -Wno-unused-member-function")
-      endif()
-      if(APPLE)
-        set(FLAGS "${FLAGS} -Wno-undef")
-        if(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER "8.0.0")
-          set(FLAGS "${FLAGS} -fcomment-block-commands=file")
-        endif()
-      else()
-        if(CMAKE_CXX_COMPILER_VERSION VERSION_EQUAL "5.0.0")
-          set(FLAGS "${FLAGS} -fcomment-block-commands=file")
-        endif()
-        if(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER "5.0.0")
-          set(FLAGS "${FLAGS} -fcomment-block-commands=file")
-        endif()
-      endif()
-    elseif(${CMAKE_CXX_COMPILER_ID} STREQUAL "GNU")
-      if(${PROJECT_NAME}_CXX_WARNINGS)
-        set(FLAGS "${FLAGS} -Wall")
-        set(FLAGS "${FLAGS} -Wcast-align")
-        set(FLAGS "${FLAGS} -Wconversion")
-        set(FLAGS "${FLAGS} -Wdouble-promotion")
-        set(FLAGS "${FLAGS} -Werror")
-        set(FLAGS "${FLAGS} -Wextra")
-        set(FLAGS "${FLAGS} -Wformat=2")
-        set(FLAGS "${FLAGS} -Wnon-virtual-dtor")
-        set(FLAGS "${FLAGS} -Wold-style-cast")
-        set(FLAGS "${FLAGS} -Woverloaded-virtual")
-        set(FLAGS "${FLAGS} -Wshadow")
-        set(FLAGS "${FLAGS} -Wsign-conversion")
-        set(FLAGS "${FLAGS} -Wunused")
-        set(FLAGS "${FLAGS} -Wuseless-cast")
-        if(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER "6.0")
-          set(FLAGS "${FLAGS} -Wduplicated-cond")
-          set(FLAGS "${FLAGS} -Wmisleading-indentation")
-          set(FLAGS "${FLAGS} -Wnull-dereference")
-        endif()
-        if(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER "7.0")
-          set(FLAGS "${FLAGS} -Wduplicated-branches")
-          set(FLAGS "${FLAGS} -Wlogical-op")
-          set(FLAGS "${FLAGS} -Wrestrict")
-        endif()
-        if(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER "8.0")
-          set(FLAGS "${FLAGS} -Wclass-memaccess")
-        endif()
-      endif()
-    elseif(${CMAKE_CXX_COMPILER_ID} STREQUAL "Intel")
-
-    else()
-      message(WARNING "Unexpected compiler type ${CMAKE_CXX_COMPILER_ID}")
-    endif()
+    bob_cxx_pedantic_flags()
     set(CMAKE_CXX_FLAGS "${FLAGS}" PARENT_SCOPE)
   endif()
 endfunction(bob_begin_cxx_flags)
 
-function(bob_cxx11_flags)
+function(bob_cxx_standard_flags standard)
   set(FLAGS "${CMAKE_CXX_FLAGS}")
-  set(FLAGS "${FLAGS} --std=c++11")
+  set(FLAGS "${FLAGS} --std=${standard}")
```

Comment:
Shouldn't this also be set via
```
set(CMAKE_CXX_STANDARD 14)
```
?

Will this also show up for exported targets in dependencies down the road?

Reply:
You are right, it should use `CMAKE_CXX_STANDARD` (appeared in CMake 3.1) instead. The reason why bob was not using it is that it originally supported older CMake versions, but hpc-coding-conventions supports CMake 3.6 or higher.
I will test the change of zee and basalt and update the branch.

Thank you for reporting this.

Reply:
46a3df4 addresses this conversation